harness/tests/tensorboard/test_base.py

```diff
@@ -90,3 +93,63 @@ def test_illegal_type() -> None:
         tensorboard.build(
             env.det_cluster_id, env.det_experiment_id, env.det_trial_id, checkpoint_config
         )
+
+
+def test_upload_thread_normal_case() -> None:
+    upload_function = mock.Mock()
+    upload_thread = tensorboard.base._TensorboardUploadThread(upload_function)
+
+    upload_thread.start()
+    upload_thread.upload(
+        [pathlib.Path("test_value/file1.json"), pathlib.Path("test_value/file2.json")]
+    )
+    upload_thread.upload([pathlib.Path("test_value/file3.json")])
+    # Pass in sentinel value to exit thread
+    upload_thread.close()
+    upload_thread.join()
+
+    assert upload_function.call_count == 2
+
+
+def test_upload_thread_exception_case() -> None:
+    # 1. Define custom hook to capture threads fail
```

Comment:
Nit: your comments are strangely reflowed.  We have 100-width lines in the python code, it looks strange to have line breaks at such a narrow width.

Reply:
Agreed, will keep it in mind for future reference.